packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/updateModal.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { X as CloseIcon } from 'react-feather';
+
+import { mergeClasses } from '../../../classify';
+import Icon from '../../Icon';
+import { Modal } from '../../Modal';
+import EditForm, { modes as editFormModes } from './editForm';
+
+import defaultClasses from './updateModal.css';
+
+const UpdateModal = props => {
+    const {
+        handleCancelUpdate,
+        handleSubmit,
+        isOpen,
+        selectedShippingMethod,
+        shippingMethods
+    } = props;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const rootClass = isOpen ? classes.root_open : classes.root;
+
+    return (
+        <Modal>
+            <aside className={rootClass}>
+                <div className={classes.header}>
+                    <span className={classes.headerText}>
+                        {'Edit Shipping Method'}
+                    </span>
+                    <button
+                        className={classes.closeButton}
+                        onClick={handleCancelUpdate}
+                    >
+                        <Icon src={CloseIcon} />
+                    </button>
+                </div>
+                <div className={classes.body}>
+                    <EditForm
+                        handleCancelUpdate={handleCancelUpdate}
+                        handleSubmit={handleSubmit}
+                        mode={editFormModes.UPDATE}
```

Comment:
@tjwiebell had a different approach to determining which state to display. He checks for the [existence of email](https://github.com/magento/pwa-studio/pull/2285/files#diff-628505350ad68f9d53340e7f89b5df0bR28) which indicates that a user has already submitted data. I'm not sure which I like better, but I just wanted to point out there is a minor difference here in the approach.

Reply:
I do a similar check for the [existence of a selected shipping method](https://github.com/magento/pwa-studio/blob/supernova/179_checkout_shipping_methods/packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js#L163-L177), but I just pass that answer to the modal in the form of a `mode` prop.

I'm not married to it but I thought it was a little more explicit in what the we wanted to show.